### DISCUSSION TITLE
1.9.x status?

### DISCUSSION THREAD

**Discussion by joakim-tjernlund** [fwupd/fwupd#5925] [Q&A]
**Describe the question**
Release train 1.8.x is described as stable but 1.9.x does not have any description.

Does this mean 1.9.x should be considered unstable?
Perhaps 1.8.x is LTS and 1.9.x is ready for public use?

**Comment by hughsie** [comment]
I'm not sure I'd call it unstable. Maybe *latest* or something.

**Comment by superm1** [comment]
Generally what we've been doing for a while is when a _new_ series opens up, the _old_ series is still supported and then we stop doing anything with previous old series.  So when `1.9.x` opened up we stopped doing anything with `1.7.x`.  Similarly I expect when we open up `2.0.x` that we'll stop supporting `1.8.x`.

Considering the above description, how about we consider 1.9.x `stable` and 1.8.x `oldstable`?

**Reply by hughsie** [reply]
> how about we consider 1.9.x stable and 1.8.x oldstable

A rose by any other name would smell as sweet. I don't mind much on the names. :)

**Reply by superm1** [reply]
I think what might be useful is to put a table on the front and center of `README.md` of the `main` branch showing something like this:

| **Series**         | **Status** |
|--------------|-----------|
| 1.9.x | latest      |
| 1.8.x | supported |
| 1.7.x | EOL as of xx/yy/zzzz |
| 1.6.x | EOL as of xx/yy/zzzz |

**Reply by hughsie** [reply]
Looks like we have https://github.com/fwupd/fwupd/blob/main/SECURITY.md -- except out of date! Want me do go find the right EOL dates?

**Reply by superm1** [reply]
Ah, good idea!  I guess the release actions list needs to add updating `SECURITY.md`. Yeah I think this would be useful for people to cross reference.

**Reply by joakim-tjernlund** [reply]
I don't think ```latest``` is a good label, it does not add any info.
If 1.9.2 is ready for wider use, please call it stable.

Gentoo maintainers thinks 1.9.x is unstable and not ready so is reluctant to add 1.9.x to the Gentoo tree.

**Reply by superm1** [reply]
In terms of how development happens, you should be able to view anything "in between" tagged releases as unstable and any tagged release as stable.  There is a pretty large array of hardware Richard has which he has semi-automated testing for when it comes to release time.  A few days before release he tests it all, and if regressions are found they're fixed.

Also there is new code two releases ago for "emulated" USB hardware.  This allows checking for regressions without the actual hardware and is run part of every single commit from CI.

**Reply by hughsie** [reply]
If it helps, what I do in Fedora was to skip 1.9.0 and 1.9.1 and then put 1.9.2 in. For RHEL I'm including 1.8.x as the update cadence is much slower, but will probably rebase to 1.9.x (1.9.8?) if a vendor shouts about needing new hardware support in RHEL. But like Mario said, 1.9.x isn't unstable.

**Reply by joakim-tjernlund** [reply]
You don't need to convince me :) but others just reading the release page.
There should be some label there that says something of its usability.

**Reply by superm1** [reply]
> but others just reading the release page.

Can you point out specifically what page people are looking at and reaching this conclusion?

> There should be some label there that says something of its usability.

I think if we drop a link on that page you're thinking of to `SECURITY.md` or make the content from `SECURITY.md` more prominent it may help.

**Reply by hughsie** [reply]
How about https://github.com/fwupd/fwupd/pull/5926

**Comment by joakim-tjernlund** [comment]
> > but others just reading the release page.
> 
> Can you point out specifically what page people are looking at and reaching this conclusion?

https://github.com/fwupd/fwupd/releases

There 1.8.x is called stable but no classification on 1.9.x

**Reply by superm1** [reply]
Ah. So either add stable to all releases or drop it from 1.8.x.

In any case; please guide them to the new security.md. this will show what's what.

**Reply by hughsie** [reply]
> Ah. So either add stable to all releases or drop it from 1.8.x.

Ya, I'll reword the next one.

**Comment by joakim-tjernlund** [comment]
> Ah. So either add stable to all releases or drop it from 1.8.x.
> 
> In any case; please guide them to the new security.md. this will show what's what.

Yes, I have pointed to this discussion. Thanks!
